### PR TITLE
CD-124: iperf driver - add target server

### PR DESCRIPTION
**Pull request by mohamedkallel**

### CODE REVIEW

File: examples/ssh/iperf3.js [hunk 1]
```diff
@@ -1,32 +1,53 @@
...
+ * 
+ * You can edit the targetServer object to configure the iPerf3 server to be used. 
+ * 
  */
 
 // Define SSH configuration
 var sshConfig = {
-    timeout: 20000
+    timeout: 60000,
+    port: 22
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.js:31]: this is the default --> you do not need it

File: examples/ssh/iperf3.js [hunk 2]
```diff
@@ -39,20 +60,20 @@ function checkSshError(err) {
...
-    return function(result){
+    return function (result) {
         var d = D.q.defer();
         sshConfig.command = command;
         D.device.sendSSHCommand(sshConfig, function (out, err) {
             if (err) {
-                checkSshError(err);
-                d.reject(err);
+                console.error(err);
+                d.resolve();
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.js:69]: Why? This will basically make all validate executions a success

File: examples/ssh/iperf3.js [hunk 3]
```diff
@@ -101,7 +122,26 @@ function validate() {
 * @documentation This procedure is used for retrieving iperf3 results
 */
 function get_status() {
-    execute()
-        .then(D.success)
-        .catch(failure);
-}
\ No newline at end of file
+    var serverIndex = 0;
+    function tryNextServer() {
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.js:126]: What is the purpose of this function? 
Just loop the severs and parametrize the conf

File: examples/ssh/iperf3.js [hunk 4]
```diff
@@ -101,7 +122,26 @@ function validate() {
 * @documentation This procedure is used for retrieving iperf3 results
 */
 function get_status() {
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.js:124]: This does not capture error cases (server not responding)

iperf3 -f m -c ping-90ms.online.net -p 5209 -R
iperf3: error - unable to connect to server: Connection refused

File: examples/ssh/iperf3.js [hunk 5]
```diff
@@ -1,32 +1,53 @@
...
+    { url: "speed.as208196.net", port: 5209 }
+];
+
 // Define the commands to be executed via SSH to retrieve speed data and variable configuration
 var execConfig = [
-    { id: "download_speed", label: "Download speed", command: "iperf3 -f m -c localhost -R | grep sender | awk -F \" \" '{print $7}'" },
-    { id: "upload_speed", label: "Upload speed", command: "iperf3 -f m -c localhost | grep sender | awk -F \" \" '{print $7}'" },
-    { id: "download_speed_udp", label: "Download speed UDP", command: "iperf3 -f m -c localhost -u -R | tail -n 4 | head -n 1 | awk -F \" \" '{print $7}'" },
-    { id: "upload_speed_udp", label: "Upload speed UDP", command: "iperf3 -f m -c localhost -u | tail -n 4 | head -n 1 | awk -F \" \" '{print $7}'" }
+    { id: "download_speed", label: "Download speed", command: "iperf3 -f m -c " + targetServers[0].url + " -p " + targetServers[0].port + " -R | grep sender | awk -F \" \" '{print $7}'" },
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.js:47]: please don't hardcode it, if you want to have multiple targets use a loop

File: examples/ssh/iperf3.json [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+        "iperf3",
+        "server",
+        "network",
+        "test",
+        "Download speed",
+        "Upload speed",
+        "Download speed UDP",
+        "Upload speed UDP"
+    ],
+    "sample_period": 60000,
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.json:25]: The sample period value is in seconds

this is 16.66667 hours, was this intentional?

File: examples/ssh/iperf3.json [hunk 7]
```diff
@@ -0,0 +1,45 @@
...
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "iperf3 installed",
+            "the server iperf3 must be running"
+        ]
+    },
+    "tags": [
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.json:15]: tags are single words (mostly) and DO NOT support upper case

Please redo them
test is meaningless speedtest is ok

File: examples/ssh/iperf3.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+        "test",
+        "Download speed",
+        "Upload speed",
+        "Download speed UDP",
+        "Upload speed UDP"
+    ],
+    "sample_period": 60000,
+    "has_actions": false,
+    "expected_variables": {
+        "table": "[2.999]"
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.json:28]: I do not see a table being created, these are key-value (independent variables)

File: examples/ssh/iperf3.json [hunk 9]
```diff
@@ -0,0 +1,45 @@
...
+    "version": "1.0.0",
+    "name": "Iperf",
+    "description": "This script is designed to retrieve network speed data using the iperf3",
+    "category": "Software Integration",
+    "requirements": {
+        "credentials": true,
+        "sandbox_version": "1.12",
+        "others": [
+            "iperf3 installed",
+            "the server iperf3 must be running"
```
Comment (GeorgiTsatsev) [inline examples/ssh/iperf3.json:12]: is this still needed?

File: examples/ssh/iperf3.js (hunk 3 above)
Comment (mohamedkallel) [reply examples/ssh/iperf3.js:126]: in the case of one server is busy we jump to the next server to test the perf

Comment (mohamedkallel) [reply examples/ssh/iperf3.js]: this section is removed, the check will be for every command server by server

**Review by GeorgiTsatsev — APPROVED** [review]

**Pull request merged**